[12.0][ADD] partner_multi_name module

Capriatto opened the pull request:
This is our first contribution, all suggestions are welcome:

This module extends the "partner_second_lastname" functionality to have a middle name or other names in a separated way.

    Use Case:
    Partner Second Last Name: <Johannes Chrysostomus>  <Theophilus> <Mozart>
    Partner Other Names: <Johannes><Chrysostomus Wolfgangus>< Theophilus><Mozart>

ping: @Yajo  @jgrandguillaume  @rafaelbn

CarlosRoca13 reviewed (approved):
LGTM :+1:

pedrobaeza commented:
A module name shouldn't contain plurals: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#modules

grottas commented:
@Capriatto partner_multi_middle_name or partner_middlename can be better module names for this one

HaraldPanten commented:
Please, check comments

rousseldenis commented:
> Please, check comments

Specifically, change to singular name

NL66278 reviewed (changes requested):
Usefull addtition and thanks for the contribution!

The name of the module must still be changed not to contain plurals. Personally I would prefer a name like partner_multi_name (without middle). Depending on country and culture you might have multiple parts in the first name, a middle name, multiple last names.

As an example taka a dutch name like Pieter Jan van Harinxma thoe Sloten (we do not actually all have such extensive names), Pieter Jan would be considered part of the last name, van would be the middle name and Harinxma thoe Sloten all part of the last name. In a telephone directory this would be formatted as: "Harinxma thoe Sloten, Pieter Jan van".

For a future version it might be better to have one module to support the splitting up of the name field, instead of several (partner_firstname, partner_second_lastname. (And we also have l10n_nl_partner_name in the Dutch localization).

Capriatto commented:
> Usefull addtition and thanks for the contribution!
> 
> The name of the module must still be changed not to contain plurals. Personally I would prefer a name like partner_multi_name (without middle). Depending on country and culture you might have multiple parts in the first name, a middle name, multiple last names.
> 
> As an example taka a dutch name like Pieter Jan van Harinxma thoe Sloten (we do not actually all have such extensive names), Pieter Jan would be considered part of the last name, van would be the middle name and Harinxma thoe Sloten all part of the last name. In a telephone directory this would be formatted as: "Harinxma thoe Sloten, Pieter Jan van".
> 
> For a future version it might be better to have one module to support the splitting up of the name field, instead of several (partner_firstname, partner_second_lastname. (And we also have l10n_nl_partner_name in the Dutch localization).

Thank you very much. We applied your suggestion of renaming the module. We also will check your localization module for a future module adaptation.

Capriatto commented:
@NL66278 the requested change is done. Thanks for your revision.

NL66278 commented on partner_multi_name/__init__.py:
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
```
No more coding lines in 12.0

Capriatto replied:
Done, thank you.

NL66278 commented on partner_multi_name/__init__.py:
```diff
@@ -0,0 +1,9 @@
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# Copyright 2018 EXA Auto Parts S.A.S Guillermo Montoya <Github@guillermm>
+# Copyright 2018 EXA Auto Parts S.A.S Joan Marín <Github@JoanMarin>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Make it https:// (also in other files).

Capriatto replied:
Done, thank you.

Capriatto commented:
@NL66278 the requested change is done. Thanks for your revision.

CarlosRoca13 commented on partner_multi_name/models/base_config_settings.py:
```diff
@@ -0,0 +1,48 @@
+    _inherit = 'base.config.settings'
+
+    def _partner_names_order_selection(self):
+        options = super(BaseConfigSettings,
+                        self)._partner_names_order_selection()
```
```suggestion
        options = super()._partner_names_order_selection()
```
In python3 you can short the call to super

CarlosRoca13 commented on partner_multi_name/models/res_partner.py:
```diff
@@ -0,0 +1,130 @@
+
+        if not is_company and name:
+            order = self._get_names_order()
+            result = super(ResPartner, self).\
+                _get_inverse_name(name, is_company)
```
```suggestion
            result = super().\
                _get_inverse_name(name, is_company)
```

CarlosRoca13 commented on partner_multi_name/models/res_partner.py:
```diff
@@ -0,0 +1,130 @@
+    def _check_name(self):
+        """Ensure at least one name is set."""
+        try:
+            super(ResPartner, self)._check_name()
```
```suggestion
            super()._check_name()
```

CarlosRoca13 commented on partner_multi_name/models/res_partner.py:
```diff
@@ -0,0 +1,130 @@
+    @api.onchange("firstname", "othernames", "lastname", "lastname2")
+    def _onchange_subnames(self):
+        """Trigger onchange with 'othernames' too."""
+        super(ResPartner, self)._onchange_subnames()
```
```suggestion
        super()._onchange_subnames()
```

CarlosRoca13 reviewed (approved):
Just non-blocking comments

CarlosRoca13 commented:
Ohh, I didn't see travis and runbot, they are red. Solve them

CarlosRoca13 commented on partner_multi_name/models/base_config_settings.py:
```diff
@@ -0,0 +1,48 @@
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
```
The problem appears with this because the model 'base.config.settings' does not exist. Did you mean 'res.config.settings'?

Capriatto replied:
you're right. Fixed.

CarlosRoca13 reviewed (approved):
LGTM :+1:

Capriatto commented:
> > Please, check comments
> 
> Specifically, change to singular name

@rousseldenis is still necessary the "needs fixing" tag?

alejo-code commented:
@NL66278    Specifically, change to singular name

JoanMarin committed:
[ADD] partner_multi_name module

alejo-code committed:
[12.0][ADD] partner_multi_name module

[ADD] partner_multi_name module

[ADD] partner_multi_name module

[ADD] partner_multi_name module

marcelsavegnago commented:
@alejo-code please squash the commits

grottas commented:
@JoanMarin @Capriatto Can you take a look at this? V 11.0 is already merged, and even V10.0 is closed as it is not maintained anymore, it has some useful [reviews](https://github.com/OCA/partner-contact/pull/859#pullrequestreview-568321159) from @janverb and @NL66278 that can be taken into account.

grottas committed:
Merge branch 'OCA:12.0' into 12.0

NL66278 reviewed (approved):
:+1: LGTM

NL66278 commented:
/ocabot merge patch
